index.d.ts

```diff
@@ -34,11 +58,61 @@ declare module 'launchdarkly-node-server-sdk-redis' {
    *   A factory function that the SDK will use to create the data store. Put this value into the
    *   `featureStore` property of [[LDOptions]].
    */
-  export default function RedisFeatureStore(
-    redisOpts?: ClientOpts,
+  export function RedisFeatureStore(
+    redisOpts?: ClientOpts | LDRedisOptions,
     cacheTTL?: number,
     prefix?: string,
     logger?: LDLogger | object,
     client?: RedisClient
   ): (config: LDOptions) => LDFeatureStore;
+
+  /**
+   * Configures a big segment store backed by a Redis instance.
+   * 
+   * "Big segments" are a specific type of user segments. For more information, read the
+   * LaunchDarkly documentation about user segments: https://docs.launchdarkly.com/home/users
+   *
+   * @param options The standard options supported for all LaunchDarkly Redis features, including both
+   *   options for Redis itself and others related to the SDK's behavior.
+   */
+  export function RedisBigSegmentStore(
+    options?: LDRedisOptions
+  ): (config: LDOptions) => ld.interfaces.BigSegmentStore;
+
+  /**
+   * The standard options supported for the LaunchDarkly Redis integration.
+   */
+  export interface LDRedisOptions {
```

Comment:
Does not include a `logger` field.

Reply:
It is not supposed to. Please see my last comment on the other thread.

Reply:
We don't want people to be specifying a separate logger for this component. That was simply an unfortunate consequence of the original decision to require people to create an _instance_ of the component themselves, which meant there was no way for it to inherit the SDK's default logger (or any other state from the SDK). Now that we are using a factory function instead that will receive the SDK's configuration, there is no reason to have a separate logger option and so it's not supported in the new config object.

Reply:
Ohh, okay. This is the piece I was missing, since I was expecting logger to be retained as something settable when calling the feature store constructors.